android/src/main/java/com/bugsnag/JavaScriptException.java

```diff
@@ -30,47 +30,108 @@ public void toStream(JsonStream writer) throws IOException {
+        // expected format is as follows:
+        //   release:
+        //     "$method@$filename:$lineNumber:$columnNumber"
+        //   dev:
+        //     "$method@?uri:$lineNumber:$columnNumber"
+
+        writer.beginObject();
+        String[] methodComponents = frame.split("@", 2);
+        String fragment = methodComponents[0];
+        if (methodComponents.length == 2) {
+            writer.name("method").value(methodComponents[0]);
+            fragment = methodComponents[1];
+        }
+
+        int columnIndex = fragment.lastIndexOf(":");
+        if (columnIndex != -1) {
+            String columnString = fragment.substring(columnIndex + 1);
+            Integer columnNumber = parseIntSafe(columnString);
+
+            if (columnNumber != null) {
+                writer.name("columnNumber").value(columnNumber);
             }
+            fragment = fragment.substring(0, columnIndex);
+        }
+
+        int lineNumberIndex = fragment.lastIndexOf(":");
+        if (lineNumberIndex != -1) {
+            String lineNumberString = fragment.substring(lineNumberIndex + 1);
+            Integer lineNumber = parseIntSafe(lineNumberString);
+
+            if (lineNumber != null) {
+                writer.name("lineNumber").value(lineNumber);
+            }
+            fragment = fragment.substring(0, lineNumberIndex);
+        }
 
-            int lineNumberIndex = fragment.lastIndexOf(":");
-            if (lineNumberIndex != -1) {
-                String lineNumberString = fragment.substring(lineNumberIndex + 1);
-                try {
-                    int lineNumber = Integer.parseInt(lineNumberString);
+        writer.name("file").value(fragment);
+        writer.endObject();
+    }
+
+    private void serialiseHermesFrame(JsonStream writer, String frame) throws IOException {
+        // expected format is as follows:
+        //   release
+        //     "at $method (address at $filename:$lineNumber:$columnNumber)"
+        //   dev
+        //     "at $method ($filename:$lineNumber:$columnNumber)"
+
+        int srcInfoStart = Math.max(frame.lastIndexOf(" "), frame.lastIndexOf("("));
+        int srcInfoEnd = frame.lastIndexOf(")");
+        boolean hasSrcInfo = srcInfoStart > -1 && srcInfoStart < srcInfoEnd;
+
+        int methodStart = "at ".length();
+        int methodEnd = frame.indexOf(" (");
+        boolean hasMethodInfo = methodStart < methodEnd;
+
+        // serialise srcInfo
+        if (hasSrcInfo || hasMethodInfo) {
+            writer.beginObject();
+            writer.name("method").value(frame.substring(methodStart, methodEnd));
+            if (hasSrcInfo) {
+                String srcInfo = frame.substring(srcInfoStart + 1, srcInfoEnd);
+                String file = srcInfo.replaceFirst(":\\d+:\\d+$", "");
+
+                writer.name("file").value(file);
+
+                String[] chunks = srcInfo.split(":");
+                Integer lineNumber = parseIntSafe(chunks[chunks.length - 2]);
```

Comment:
We should check the size of `chunks.length` here, just to be defensive about passing in a negative index into an array